legend-sdlc-server-fs/src/main/java/org/finos/legend/sdlc/server/api/project/FileSystemProjectApi.java

```diff
@@ -0,0 +1,308 @@
+
+            repository.getConfig().setString("project", null, "id", projectId);
+            repository.getConfig().setString("project", null, "name", name);
+            repository.getConfig().setString("project", null, "description", description);
+            repository.getConfig().save();
+        }
+        catch (IOException | GitAPIException e)
+        {
+            LOGGER.error("Failed to create project {}", name, e);
+        }
+        if (gitProject == null)
+        {
+            throw new LegendSDLCServerException("Failed to create project: " + name);
+        }
+        Project project = gitProjectToProject(gitProject);
+
+        // Build project structure
+        int projectStructureVersion = getDefaultProjectStructureVersion();
+        ProjectConfigurationUpdater configUpdater = ProjectConfigurationUpdater.newUpdater()
+                .withProjectId(project.getProjectId())
+                .withProjectType(type)
+                .withGroupId(groupId)
+                .withArtifactId(artifactId)
+                .withProjectStructureVersion(projectStructureVersion);
+        if (this.projectStructureExtensionProvider != null && type != ProjectType.EMBEDDED)
+        {
+            configUpdater.setProjectStructureExtensionVersion(this.projectStructureExtensionProvider.getLatestVersionForProjectStructureVersion(projectStructureVersion));
+        }
+        ProjectStructure.newUpdateBuilder(FileSystemProjectApi.getProjectFileAccessProvider(), project.getProjectId(), configUpdater)
+                .withMessage("Build project structure")
+                .withProjectStructureExtensionProvider(this.projectStructureExtensionProvider)
+                .withProjectStructurePlatformExtensions(this.projectStructurePlatformExtensions)
+                .build();
+        return project;
+    }
+
+    public static ProjectFileAccessProvider getProjectFileAccessProvider()
+    {
+        return new ProjectFileAccessProvider()
+        {
+            @Override
+            public FileAccessContext getFileAccessContext(String projectId, SourceSpecification sourceSpecification, String revisionId)
+            {
+                return new FileSystemEntityApi.AbstractFileSystemFileAccessContext(projectId, sourceSpecification, revisionId);
+            }
+
+            @Override
+            public RevisionAccessContext getRevisionAccessContext(String projectId, SourceSpecification sourceSpecification, Iterable<? extends String> paths)
+            {
+                return new FileSystemEntityApi.FileSystemRevisionAccessContext(projectId, sourceSpecification, paths);
+            }
+
+            @Override
+            public FileModificationContext getFileModificationContext(String projectId, SourceSpecification sourceSpecification, String revisionId)
+            {
+                return new FileSystemEntityApi.FileSystemProjectFileFileModificationContext(projectId, sourceSpecification, revisionId);
+            }
+        };
+    }
```

Comment:
It seems like these classes should be top level classes instead of nested classes in FileSystemEntityApi. Also, it seems like AbstractFileSystemFileAccessContext is misnamed, since it is apparently not abstract.

Perhaps this static method should also be moved to a central location instead of residing in FileSystemProjectApi?